x/blob/client/cli/payforblob.go

```diff
@@ -63,11 +65,21 @@ func CmdPayForBlob() *cobra.Command {
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
+	cmd.PersistentFlags().Uint8(FlagNamespaceVersion, 0, "Specify the namespace version")
 	cmd.PersistentFlags().Uint8(FlagShareVersion, 0, "Specify the share version")
-
 	return cmd
 }
 
+func getNamespace(namespaceID []byte, namespaceVersion uint8) (appns.Namespace, error) {
+	switch namespaceVersion {
+	case appns.NamespaceVersionZero:
+		namespaceVersionZeroPrefix := appns.NamespaceVersionZeroPrefix
+		return appns.New(namespaceVersion, append(namespaceVersionZeroPrefix, namespaceID...))
```

Comment:
this doesn't actually avoid the issue, since slices can be treated similar to pointers. Its true that we're passing by value here, but we're passing a pointer by value, which unfortunatley still won't solve our problem

in order to avoid the underlying issue, we need to make an explicit copy of the slice
similar to 
```go
ns := make([]byte, 0, appconsts.NamespaceSize)
ns = append(ns, append(prefix, namespaceID)...)
```

Reply:
I guess now I get it.
with above example prefix may still be changed
```
nsp := make([]byte, 0, appns.NamespaceVersionZeroPrefixSize)
nsp = append(nsp, appns.NamespaceVersionZeroPrefix...)
return appns.New(namespaceVersion, append(nsp, namespaceID...))
```
I have updated the code to this